### PR TITLE
add healthmonitor function

### PR DESCRIPTION
**Pull request by greatbn**

### CODE REVIEW

File: loadbalancer.go [hunk 1]
```diff
@@ -732,3 +731,154 @@ func (p *pool) Delete(ctx context.Context, id string) error {
...
+	Name           string                `json:"name"`
+	Type           string                `json:"type"`
+	Delay          int                   `json:"delay"`
+	MaxRetries     int                   `json:"max_retries"`
+	MaxRetriesDown int                   `json:"max_retries_down"`
+	TimeOut        int                   `json:"timeout"`
+	HTTPMethod     string                `json:"http_method"`
+	UrlPath        string                `json:"url_path"`
+	ExpectedCodes  string                `json:"expected_codes"`
+	HTTPVersion    float32                `json:"http_version"`
```
Comment (cuonglm) [inline loadbalancer.go:745]: `gofmt`

File: loadbalancer.go [hunk 2]
```diff
@@ -732,3 +731,154 @@ func (p *pool) Delete(ctx context.Context, id string) error {
...
+
+type HealthMonitorUpdateRequest struct {
+	Name           string `json:"name"`
+	TimeOut        string `json:"timeout"`
+	PoolID         string `json:"pool_id"`
+	Delay          int    `json:"delay"`
+	MaxRetries     int    `json:"max_retries"`
+	MaxRetriesDown int    `json:"max_retries_down"`
+	HTTPMethod     string `json:"http_method"`
+	HTTPVersion    float32 `json:"http_version"`
```
Comment (cuonglm) [inline loadbalancer.go:778]: `gofmt`

File: loadbalancer.go [hunk 3]
```diff
@@ -732,3 +731,154 @@ func (p *pool) Delete(ctx context.Context, id string) error {
...
+type HealthMonitorCreateRequest struct {
+	Name           string `json:"name"`
+	Type           string `json:"type"`
+	TimeOut        int    `json:"timeout"`
+	PoolID         string `json:"pool_id"`
+	Delay          int    `json:"delay"`
+	MaxRetries     int    `json:"max_retries"`
+	MaxRetriesDown int    `json:"max_retries_down"`
+	HTTPMethod     string `json:"http_method"`
+	HTTPVersion    float32 `json:"http_version"`
```
Comment (cuonglm) [inline loadbalancer.go:764]: `gofmt`

File: loadbalancer.go [hunk 4]
```diff
@@ -732,3 +731,154 @@ func (p *pool) Delete(ctx context.Context, id string) error {
...
+	UrlPath        string                `json:"url_path"`
+	ExpectedCodes  string                `json:"expected_codes"`
+	HTTPVersion    float32                `json:"http_version"`
+	OpratingStatus string                `json:"oprating_status"`
+	DomainName     string                `json:"domain_name"`
+	ID             string                `json:"id"`
+	CreatedAt      string                `json:"created_at"`
+	UpdatedAt      string                `json:"updated_at"`
+	TenantID       string                `json:"tenant_id"`
+	Pool           []struct{ ID string } `json:"pool"`
```
Comment (cuonglm) [inline loadbalancer.go:752]: This `[]struct{ ID string }` is used in many places, maybe we should add new type for this dedicated type.

Comment (greatbn) [reply loadbalancer.go:752]: Yes.

**Pull request merged**